hydrolib/core/basemodel.py

```diff
@@ -646,13 +696,20 @@ def __init__(
         Args:
             filepath (Optional[Path], optional): The file path. Defaults to None.
             resolve_casing (bool, optional): Whether or not to resolve the file name references so that they match the case with what is on disk. Defaults to False.
+            recurse (bool): Whether or not to recursively load the model. Default to True.
         """
         if not filepath:
             super().__init__(*args, **kwargs)
             return
 
         with file_load_context() as context:
-            context.set_resolve_casing(resolve_casing)
+            context.initialize_load_settings(recurse, resolve_casing)
+
+            if not context.load_settings.recurse and not context.cache_is_empty():
```

Comment:
Perhaps we can add a comment or put this in a separate function as it may not be clear at first glance when we should go into this if-statement.